supervisor/shared/filesystem.c

```diff
@@ -199,5 +201,12 @@ void filesystem_set_concurrent_write_protection(fs_user_mount_t *vfs, bool concu
 }
 
 bool filesystem_present(void) {
-    return true;
```

Comment:
It's a very unlikely scenario but a device could have damaged flash but someone mounts an sd filesystem via the repl. This would make this function return true but the internal filesystem object would be invalid.

The rest of the PR is great though and this is an unlikely story so don't let this alone stop it being merged

Reply:
@tannewt is it worth checking for this? Right now you go into safe mode if `filesystem_init()` fails for CIRCUITPY.

Reply:
I think I will merge anyway to get this in and an issue could be opened if it's worth it.

Reply:
_mp_vfs.len isn't the length of vfs mounts. It's the length of the mount point string for the "native" filesystems vfs entry. So I *think* it'll be ok even when only an SD card exists.